core/src/main/java/info/nightscout/androidaps/plugins/pump/common/defs/PumpType.kt

```diff
@@ -89,7 +89,7 @@ enum class PumpType {
         baseBasalMinValue = 0.02,
         baseBasalMaxValue = null,
         baseBasalStep = 0.01,
-        baseBasalSpecialSteps = DoseStepSize.InsightBolus,
+        baseBasalSpecialSteps = DoseStepSize.InsightBasal,
```

Comment:
@TebbeUbben This change is untested (due to lack of physical pump), I just saw this while trying to understand what's causing the issue here: https://github.com/nightscout/AndroidAPS/pull/472#discussion_r620451352

Reply:
I still have the same exception (insulin = 0.0 and carbs = 0.0)

Reply:
I tried to find from where comes the call in debug : CommandSMBBolus (line 30)
```
    override fun execute() {
        val r: PumpEnactResult
        val lastBolusTime = repository.getLastBolusRecord()?.timestamp ?: 0L
        if (lastBolusTime != 0L && lastBolusTime + T.mins(3).msecs() > dateUtil.now()) {
            aapsLogger.debug(LTag.PUMPQUEUE, "SMB requested but still in 3 min interval")
            r = PumpEnactResult(injector).enacted(false).success(false).comment("SMB requested but still in 3 min interval")
        } else if (detailedBolusInfo.deliverAtTheLatest != 0L && detailedBolusInfo.deliverAtTheLatest + T.mins(1).msecs() > System.currentTimeMillis()) {
            r = activePlugin.activePump.deliverTreatment(detailedBolusInfo)     // <=== call from here
        } else {
            r = PumpEnactResult(injector).enacted(false).success(false).comment("SMB request too old")
            aapsLogger.debug(LTag.PUMPQUEUE, "SMB bolus canceled. deliverAt: " + dateUtil.dateAndTimeString(detailedBolusInfo.deliverAtTheLatest))
        }
        aapsLogger.debug(LTag.PUMPQUEUE, "Result success: ${r.success} enacted: ${r.enacted}")
        callback?.result(r)?.run()
    }
```

Then from QueueThread (line 118)

I don't know if this could help...

Reply:
Now I remember, when I worked on userEntru I noticed that sometimes LoopPlugin had request with 0.0 insulin (I filtered these userentries)
See `LoopPlugin` line 627 below
```
        // deliver SMB
        val detailedBolusInfo = DetailedBolusInfo()
        detailedBolusInfo.lastKnownBolusTime = repository.getLastBolusRecord()?.timestamp ?: 0L
        detailedBolusInfo.eventType = DetailedBolusInfo.EventType.CORRECTION_BOLUS
        detailedBolusInfo.insulin = request.smb
        detailedBolusInfo.bolusType = DetailedBolusInfo.BolusType.SMB
        detailedBolusInfo.deliverAtTheLatest = request.deliverAt
        aapsLogger.debug(LTag.APS, "applyAPSRequest: bolus()")
        if (request.smb > 0.0)
            uel.log(Action.SMB, Sources.Loop, ValueWithUnit.Insulin(detailedBolusInfo.insulin))
        commandQueue.bolus(detailedBolusInfo, callback)
```

Reply:
My suggestion to solve this is to complete the test at the beginning of applySMBRequest with smb value:
```
    private fun applySMBRequest(request: APSResult, callback: Callback?) {
        if (!request.bolusRequested || request.smb == 0.0) {
            return
        }
```

Reply:
That solution would just hide the real problem since `bolusRequested` is derived from `smb`. I found the issue causing this: https://github.com/nightscout/AndroidAPS/pull/498